### DISCUSSION TITLE
What would you like to see in V4? :star2:

### DISCUSSION THREAD

**Discussion by tinchoabbate** [tinchoabbate/damn-vulnerable-defi#51] [Help / Q&A / Troubleshooting] [answered]
I'm working on the V4 release of Damn Vulnerable DeFi! What new features and challenges would you like to see?

Examples of things I'm already working on:

- Upgrading all dependencies to latest versions (OZ Contracts v5, latest Safe stuff, etc)
- Full migration to Foundry
- New challenges with issues like flawed token distributions with merkle trees or account impersonation with meta txs

Drop below whatever comes to mind! Think of new / popular libraries, scenarios, use cases, contracts, issues, ways of playing, and more.

**Comment by letonchanh** [comment]
Love the migration to Foundry.

**Comment by matrix-0wl** [comment]
Great! When can we expect the public release of the version in Foundry?

**Reply by tinchoabbate** [reply]
august **_maybe_**

**Comment by tinchoabbate** [comment, marked as answer]
[With v4 released](https://www.damnvulnerabledefi.xyz/v4-release/), I'm closing this now. Thanks people!
